src/middleware/pageBuilder.js

```diff
@@ -0,0 +1,35 @@
+import * as t from '@/actions/types';
+import * as selectors from '@/reducers/selectors';
+import * as actions from '@/actions/creators';
+
+export default ({ dispatch, getState }) => next => action => {
+  const { type, payload } = action;
+
+  if (type === t.ADD_PAGE) {
+    const pageType = payload.type || 'standard';
+    const fields = selectors.getPageFieldNames(getState(), pageType);
+    const protectedSections = selectors.getProtectedSectionsForPage(getState(), pageType);
+    const minSectionsPerPage = selectors.getMinSectionsPerPage(getState());
+
+    // create the new page
+    const pageId = Math.random().toString(36).substring(2, 9);
```

Comment:
The ID generator should have it's own helper file so that we can replace it easily later.